Orleans.Providers.MongoDB/Reminders/Store/MongoReminderCollection.cs

```diff
@@ -35,48 +37,66 @@ protected override string CollectionName()
 
         protected override void SetupCollection(IMongoCollection<MongoReminderDocument> collection)
         {
-            var byHashDefinition =
+            var byGrainHashDefinition =
                 Index
-                    .Ascending(x => x.IsDeleted)
                     .Ascending(x => x.ServiceId)
                     .Ascending(x => x.GrainHash);
             try
             {
                 collection.Indexes.CreateOne(
-                    new CreateIndexModel<MongoReminderDocument>(byHashDefinition,
+                    new CreateIndexModel<MongoReminderDocument>(byGrainHashDefinition,
                         new CreateIndexOptions
                         {
-                            Name = "ByHash"
+                            Name = "ByGrainHash"
                         }));
             }
             catch (MongoCommandException ex)
             {
-                if (ex.CodeName == "IndexOptionsConflict")
+                if (!ex.IsDuplicateIndex())
                 {
-                    collection.Indexes.CreateOne(new CreateIndexModel<MongoReminderDocument>(byHashDefinition));
+                    throw;
```

Comment:
Same question about re-throwing? Do we want failures here to potentially kill the silo?